packages/core/src/codewhispererChat/controllers/chat/controller.ts

```diff
@@ -111,8 +113,10 @@ export class ChatController {
 
         onDidChangeAmazonQVisibility(visible => {
             if (visible) {
+                void globals.context.globalState.update(isAmazonQVisibleKey, true)
                 this.telemetryHelper.recordOpenChat()
             } else {
+                void globals.context.globalState.update(isAmazonQVisibleKey, false)
```

Comment:
We have too many globals. This is yet another. Why is this needed? It looks like the state could be stored in this class, instead of globalState. Is the intention to communicate with other instances of vscode at the same time? If so, that should be called out in a comment.